src/pages/projects/actions/ProjectUsedByList.tsx

```diff
@@ -0,0 +1,186 @@
+              to={`/ui/project/${encodeURIComponent(projectName)}/profiles/${encodeURIComponent(profile)}`}
+            />
+          </div>
+        )),
+      },
+    ],
+  };
+};
+
+const imagesRow = (images: string[]): MainTableRow | null => {
+  if (!images.length) {
+    return null;
+  }
+
+  return {
+    columns: [
+      {
+        content: (
+          <>
+            <Icon name="image" className="icon" />
+            Images ({images.length})
+          </>
+        ),
+      },
+      {
+        content: images.map((image) => <div key={image}>{image}</div>),
+      },
+    ],
+  };
+};
+
+const volumesRow = (volumes: string[]): MainTableRow | null => {
+  if (!volumes.length) {
+    return null;
+  }
+  return {
+    columns: [
+      {
+        content: (
+          <>
+            <Icon name="switcher-dashboard" className="icon" />
+            Volumes ({volumes.length})
+          </>
+        ),
+      },
+      {
+        content: volumes.map((volume) => <div key={volume}>{volume}</div>),
+      },
+    ],
+  };
+};
+
+interface Props {
+  project: LxdProject;
+}
+
+const ProjectUsedByList: FC<Props> = ({ project }) => {
+  const { data: imagesFromProject = [] } = useImagesInProject(project.name);
+  const resourceTypes = ["instance", "profile", "image", "volume"];
+
+  const rows = resourceTypes.map((resourceType) => {
+    const usedBy = filterUsedByType(
+      resourceType as ResourceType,
+      project.used_by?.filter(
+        // the default profile is not blocking project deletion and can't be removed itself
+        (item) => !item.startsWith("/1.0/profiles/default"),
+      ),
+    );
+
+    return {
+      type: resourceType as ResourceType,
+      count: usedBy.length,
+      items: usedBy,
+    };
+  });
+
+  const getItems = (
+    rows: {
+      type: ResourceType;
+      count: number;
+      items: { name: string; project: string; instance?: string }[];
+    }[],
+    resourceType: string,
+  ) => {
+    return rows
+      .filter((t) => t.type === resourceType && t.count > 0)
+      .flatMap((t) => t.items)
+      .map((t) => t.name);
+  };
+
+  const instances = getItems(rows, "instance");
+  const profiles = getItems(rows, "profile");
+  const images = getItems(rows, "image")
+    .map((fingerprint) =>
+      imagesFromProject.find((img) => img.fingerprint === fingerprint),
+    )
+    .filter((t) => !!t)
+    .map((t) => t.properties?.description || "");
+  const volumes = getItems(rows, "volume");
+
+  const tableRows = useMemo(() => {
+    return [
+      instancesRow(instances, project.name),
+      profilesRow(profiles, project.name),
+      imagesRow(images),
+      volumesRow(volumes),
+    ].filter((row): row is Exclude<typeof row, null> => row !== null);
+  }, [instances, profiles, images, volumes]);
+
+  return (
+    <>
+      <MainTable
+        headers={[{ content: "type" }, { content: "name" }]}
+        rows={tableRows}
+        className="p-main-table delete-project-table"
+      />
+    </>
```

Comment:
We don't need the fragment, but my main concern is: It might be simpler to rely on a raw table instead of using the main table. See how the used by list on a storage pool is built, like in [this example](https://lxd-ui-1600.demos.haus/ui/project/pr1/storage/pool/default). Or check `StorageUsedBy.tsx`, which might be more react like and does the collapsing if more than 5 of one entity type are present.